packages/aws-cdk-lib/aws-apigatewayv2/lib/websocket/integration.ts

```diff
@@ -253,6 +261,73 @@ export abstract class WebSocketRouteIntegration {
   public abstract bind(options: WebSocketRouteIntegrationBindOptions): WebSocketRouteIntegrationConfig;
 }
 
+/**
+ * The abstract class that all two-way communication route integration classes will implement.
+ */
+export abstract class WebSocketTwoWayRouteIntegration extends WebSocketRouteIntegration {
+  private responses: InternalWebSocketIntegrationResponseProps[] = [];
+
+  /**
+   * Initialize an integration for a route on websocket api.
+   * @param id id of the underlying `WebSocketIntegration` construct.
+   */
+  constructor(id: string) {
+    super(id);
+  }
+
+  /**
+   * Internal method called when binding this integration to the route.
+   * @internal
+   */
+  public _bindToRoute(options: WebSocketRouteIntegrationBindOptions): { readonly integrationId: string } {
+    const requiresBinding = !this.integration;
+    const result = super._bindToRoute(options);
+
+    if (requiresBinding) {
+      // This should never happen, super._bindToRoute must have set up the integration
+      if (!this.config || !this.integration) {
+        throw new Error('Missing integration setup during WebSocketRouteIntegration._bindToRoute');
+      }
+
+      this.responses.push(...this.config.responses ?? []);
+      if (this.responses.length && !options.returnResponse) {
+        // FIXME change to a warning?
+        throw new Error('Setting up integration responses without setting up returnResponse to true will have no effect, and is likely a mistake.');
```

Comment:
Not sure which is best here, as far as I can tell the web console (and thus I assume Cfn) allows the user to create responses, with a simple notice:

<img width="1011" alt="image" src="https://github.com/aws/aws-cdk/assets/2505696/4e88ff71-428b-40d1-94c1-de846a790c37">

It does feel like quite a big pitfall though, I don't think an exception is unwarranted

Reply:
I can see a stronger reason to allow this in the web console, say if you want to add responses first, before enabling `returnResponse`. With IaC, you can just block comment both the integration responses and the property. I don't mind just showing a warning, an exception just seemed like it would save people time debugging something that could could be easily missed at deploy time